docs/callbacks/summarize_images.rst

```diff
@@ -0,0 +1,261 @@
+      * Input layer name
+
+        String to specify the name of the input layer with the
+        original images.
+
+        .. code-block:: c
+
+           std::string input_layer_name
+
+      .. _num-tracked-images:
+      * Number of tracked images
+
+        Integer to set the number of images to track. The default
+        value is 10.
+
+        .. code-block:: c
+
+           size_t num_images = 10
+
+    .. _categorical-accuracy:
+    - Categorical Accuracy
+
+      This strategy is used to view a snapshot of images in the
+      dataset being used in the training session. It can display a
+      given number of images based on the criteria *matched correct
+      category*, *did not match correct category*, or *all images*.
+
+      Each epoch, all images that match the selected criteria are
+      outputted until the specified number of images is reached, or
+      there are no more images that match the criteria.
+
+       .. _accuracy-layer-name:
+       * Categorical accuracy layer name
+
+         String to specify the name of the categorical accuracy layer.
+         Defined in the Python front end.
+
+         .. code-block:: c
+
+            std::string cat_accuracy_layer_name
+
+       .. _selection-criteria:
+       * Image selection criteria
+
+         Criteria for selecting images to output: correct (MATCH),
+         incorrect (NOMATCH), or all images (ALL).
+
+         .. note:: *In Python front end, enum values from
+                   'lbann.CategoricalAccuracyStrategy.MatchType'
+                   must be used. The options are 'NOMATCH',
+                   'MATCH', or 'ALL'. The default value is
+                   'NOMATCH'.*
+
+         .. code-block:: c
+
+            MatchType match_type=NOMATCH
+
+       .. _num-images-per-epoch:
+       * Number of images per epoch
+
+         Number of images to output per epoch. The default value is 10.
+
+         .. code-block:: c
+
+            size_t num_images=10
+
+.. _image-source-layer:
++ **Image source layer name**
+
+  String to specify which layer to pull images from. Defined in the
+  Python Front end.
+
+  .. note:: *For the track-sample-ids strategy, the image source layer
+            must be the reconstruction layer or the dropout layer.*
+
+  .. code-block:: c
+
+     std::string img_source_layer_name
+
+.. _epoch-interval:
++ **Epoch interval**
+
+  Integer that sets the frequency to output images. The default value
+  is 1, i.e., per every 1 epoch.
+
+  .. code-block:: c
+
+     uint64_t interval = 1
+
+.. _image-format:
++ **Image format**
+
+  String to specify the image format. The default is .jpg.
+
+  .. code-block:: c
+
+     std::string img_format = ".jpg"
+
+---------------------------------------------
+Examples Using Summarize Images Callback
+---------------------------------------------
+
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Python front end
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
++ Track Sample IDs Strategy
+
+  .. code-block:: python
```

Comment:
For both the PFE and the proto examples, I would put in the sample layer definitions before referring to the names so that it's clear that, e.g., "top1" refers to a categorical accuracy layer.